pkg/block/adapter.go

```diff
@@ -221,3 +220,43 @@ type Adapter interface {
 
 	RuntimeStats() map[string]string
 }
+
+type WalkerOptions struct {
+	StorageURI     *url.URL
+	SkipOutOfOrder bool
+}
+
+type WalkerWrapper struct {
+	walker Walker
+	uri    *url.URL
+}
+
+func NewWalkerWrapper(walker Walker, uri *url.URL) *WalkerWrapper {
+	return &WalkerWrapper{
+		walker: walker,
+		uri:    uri,
+	}
+}
```

Comment:
Why is this needed? I only see calls from tests

Reply:
Only for tests